### PR TITLE
KAFKA-7741: streams-scala - document dependency workaround

### PR DESCRIPTION
**Pull request by vvcephei**
Document the workaround detailed in https://issues.apache.org/jira/browse/KAFKA-7741

### CODE REVIEW

**Comment by vvcephei** [comment]
@guozhangwang @mjsax @bbejeck, Please take a look at this when you get a chance.

Here's the rendered view of the 2.0 change:
notes-20

and the 2.1 change:
notes-21

File: 20/streams/developer-guide/dsl-api.html [hunk 1]
```diff
@@ -3290,6 +3290,32 @@ <h4><a id="streams_concepts_globalktable" href="#streams_concepts_globalktable">
             <pre class="brush: scala;">
               libraryDependencies += "org.apache.kafka" %% "kafka-streams-scala" % "{{fullDotVersion}}"
             </pre>
+            <p>
+              <strong>Notes</strong>:
+              <ul>
+                <li>
+                  The bugfix version <code>2.0.1</code> fixed several important flaws in  <code class="docutils literal"><span class="pre">kafka-streams-scala</span></code>.
+                  It's strongly recommended to use the latest bugfix release in general and to avoid <code>2.0.0</code> specifically
```
Comment (bbejeck) [inline 20/streams/developer-guide/dsl-api.html]: nit: `avoid 2.0.0` -> `avoid the 2.0.0 version of  kafka-streams-scala`

**Review by bbejeck — APPROVED** [review]
Thanks for the doc update, @vvcephei.  Just one minor nit otherwise LGTM

File: 20/streams/developer-guide/dsl-api.html (hunk 1 above)
Comment (vvcephei) [reply 20/streams/developer-guide/dsl-api.html]: Thanks. I'll make this change in all the PRs after I finish collecting reviews.

**Comment by vvcephei** [comment]
Took your suggestion @bbejeck . Thanks again.

**Review by guozhangwang — APPROVED** [review]

**Pull request merged**